app/assets/javascripts/components/user_photo_edit.es6.jsx

```diff
@@ -0,0 +1,201 @@
+/**
+ * @prop user - user
+ * @prop images - photo gallery for user
+**/
+class UserPhotoEdit extends React.Component {
+constructor(props) {
+    super(props);
+    this._openModal = this._openModal.bind(this);
+    this._cancelEdit = this._cancelEdit.bind(this);
+    this._enableEdit = this._enableEdit.bind(this);
+    this._disableEdit = this._disableEdit.bind(this);
+    this._selectImage = this._selectImage.bind(this);
+    this._handleFileChange = this._handleFileChange.bind(this);
+    this._handleTextChange = this._handleTextChange.bind(this);
+    this._renderImages = this._renderImages.bind(this);
+    this._saveForm = this._saveForm.bind(this);
+    this._successfulSave = this._successfulSave.bind(this);
+    this.state = {
+      images: this.props.images,
+      photo_ids: [],
+      showModal: false,
+      showEditForm: false,
+      title: "",
+      description: "",
+      photo_data: "",
+      render_photo: "",
```

Comment:
nit: make this camel case "renderPhoto" (props passed in from rails backend are generally snake_case, but then state data in react is usually camelCase)